tests/eas/acceptance.py

```diff
@@ -378,193 +378,65 @@ class WakeMigration(unittest.TestCase):
-                    .format(switch_window))
-
-    def test_little_big_switch2(self):
+    @experiment_test
+    def test_little_big_switch2(self, experiment, tasks):
         """Wake Migration: LITTLE -> BIG: 2"""
 
         # little - big - little - big
         #                       ^
         # We want to test that this little to big migration happens.  So we skip
         # the first three phases.
-        expected_time = self.offset + 3 * self.phase_duration
-        switch_window = (
-            expected_time -
-            SWITCH_WINDOW_HALF,
-            expected_time +
-            SWITCH_WINDOW_HALF)
-
-        logging.info(
-            "Wake Migration: LITTLE -> BIG Window: {}".format(switch_window))
-
-        log_result(
-            self.s_assert.assertSwitch(
-                "cluster",
-                self.env.target.bl.littles,
-                self.env.target.bl.bigs,
-                window=switch_window), self.log_fh)
+        self._assert_switch(experiment, "big", 3)
 
-        self.assertTrue(
-            self.s_assert.assertSwitch(
-                "cluster",
-                self.env.target.bl.littles,
-                self.env.target.bl.bigs,
-                rank=self.num_tasks,
-                window=switch_window),
-            msg="Not all tasks are wake-migrated to big cores in the expected window: {}"\
-                    .format(switch_window))
-
-    def test_big_little_switch1(self):
+    @experiment_test
+    def test_big_little_switch1(self, experiment, tasks):
         """Wake Migration: BIG -> LITLLE: 1"""
-        expected_time = self.offset
-        switch_window = (
-            max(expected_time - SWITCH_WINDOW_HALF, 0), expected_time + SWITCH_WINDOW_HALF)
+        self._assert_switch(experiment, "little", 0)
 
-        logging.info(
-            "Wake Migration: BIG -> LITTLE Window: {}".format(switch_window))
-
-        log_result(
-            self.s_assert.assertSwitch(
-                "cluster",
-                self.env.target.bl.bigs,
-                self.env.target.bl.littles,
-                window=switch_window), self.log_fh)
-
-        self.assertTrue(
-            self.s_assert.assertSwitch(
-                "cluster",
-                self.env.target.bl.bigs,
-                self.env.target.bl.littles,
-                rank=self.num_tasks,
-                window=switch_window),
-            msg="Not all tasks are wake-migrated to LITTLE cores in the expected window: {}"\
-                    .format(switch_window))
-
-    def test_big_little_switch2(self):
+    @experiment_test
+    def test_big_little_switch2(self, experiment, tasks):
         """Wake Migration: BIG -> LITLLE: 2"""
 
         # little - big - little - big
         #              ^
         # We want to test that this big to little migration happens.  So we skip
         # the first two phases.
-        expected_time = self.offset + 2 * self.phase_duration
```

Comment:
Agreed, will split this patch into one that ports to EasTest and one that factors out the commonality between the switch tests.